test/unit/transports/ssh_test.rb

```diff
@@ -46,23 +46,38 @@
     let(:ssh) { cls.new({ host: 'dummy' }) }
     let(:opts) { { } }
     let(:connection_options) { ssh.send(:connection_options, opts) }
+    let(:old_net_ssh) { 
+      current_version = Net::SSH::Version::CURRENT
+      threshold_version = Net::SSH::Version[5, 0, 0]
+      current_version < threshold_version
+    }
 
     it 'does not set a paranoid option - deprecated in net-ssh 4.2' do
       connection_options.key?(:paranoid).must_equal false
     end
 
     it 'defaults verify_host_key option to false, and does not set paranoid' do
-      connection_options[:verify_host_key].must_equal false
+      expected = old_net_ssh ? false : :never
+      connection_options[:verify_host_key].must_equal expected
```

Comment:
Now that I'm re-reading this, I'm thinking separate tests that force the version behavior and verify each separately would be better.